compute/ibm_vpc.py

```diff
@@ -0,0 +1,451 @@
+import logging
+import re
+import socket
+from datetime import datetime, timedelta
+from time import sleep
+from typing import List
+
+from ibm_cloud_networking_services import DnsSvcsV1
+from ibm_cloud_sdk_core.api_exception import ApiException
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_vpc import VpcV1
+
+from ceph.parallel import parallel
+
+LOG = logging.getLogger()
+
+socket.setdefaulttimeout(280)
+
+
+def get_ibm_service(accessKey: str):
+    """
+    Get ibm service
+    Args:
+        accessKey    The access key(API key) of the user.
+    """
+    authenticator = IAMAuthenticator(accessKey)
+    service = VpcV1(authenticator=authenticator)
+    service.set_service_url("https://au-syd.iaas.cloud.ibm.com/v1")
+    return service
+
+
+def get_dns_service(accessKey: str):
+    """
+    Get dns service
+    Args:
+        accessKey    The access key(API key) of the user.
+    """
+    authenticator = IAMAuthenticator(accessKey)
+    dnssvc = DnsSvcsV1(authenticator=authenticator)
+    dnssvc.set_service_url("https://api.dns-svcs.cloud.ibm.com/v1")
+    return dnssvc
+
+
+# Custom exception objects
+class ResourceNotFound(Exception):
+    pass
+
+
+class ExactMatchFailed(Exception):
+    pass
+
+
+class VolumeOpFailure(Exception):
+    pass
+
+
+class NetworkOpFailure(Exception):
+    pass
+
+
+class NodeError(Exception):
+    pass
+
+
+class NodeDeleteFailure(Exception):
+    pass
+
+
+class CephVMNodeV2:
+    """Represent the VMNode required for cephci."""
+
+    def __init__(self, accessKey: str, **kwargs) -> None:
+        """
+        Initialize the instance using the provided information.
+
+        Args:
+            accessKey    The access key(API key) of the user.
+        """
+        self.service = get_ibm_service(accessKey)
+        self.node = None
+
+        # CephVM attributes
+        self._subnet: list = list()
+        self._roles: list = list()
+
+    def get_id_by_name(self, name_of_resource, json_obj):
+        """
+        Gets ID of the resource when name of the resource is given
+        Args:
+            name_of_resource    Name of the resource.
+            json_obj            json object
+        """
+        resource_url = json_obj["first"]["href"]
+        resource_list_name = re.search(r"v1/(.*?)\?", resource_url).group(1)
+        resource_id = [
+            i["id"]
+            for i in json_obj[resource_list_name]
+            if i["name"] == name_of_resource
+        ][0]
+        return resource_id
```

Comment:
Let's not traverse through the complete list. Instead return as soon as a match is found.